firebase_admin/_user_mgt.py

```diff
@@ -372,6 +374,81 @@ def photo_url(self):
     def provider_id(self):
         return self._data.get('providerId')
 
+class ActionCodeSettings(object):
+    """Contains required continue/state URL with optional Android and iOS settings.
+    Used when invoking the email action link generation APIs.
+    """
+    KEYS = set(['url', 'handle_code_in_app', 'dynamic_link_domain', 'ios_bundle_id',
+                'android_package_name', 'android_minimum_version', 'android_install_app'])
+
+    def __init__(self, data=None):
+        super(ActionCodeSettings, self).__init__()
+        data = data or {}
+        if not isinstance(data, dict):
+            raise ValueError('Invalid data argument: {0}. Must be a dictionary.'.format(data))
+        if len(set(six.iterkeys(data)) - self.KEYS):
+            raise ValueError('Invalid settings provided: {0}. Valid dictionary \
+                have following keys {1}'.format(data, self.KEYS))
+        self._data = copy.deepcopy(data)
+
+    @property
```

Comment:
Define and assign attributes in the constructor. Follow the same pattern used in the FCM API in `_messaging_utils.py`.